cmake/dd_profiling.version

```diff
@@ -0,0 +1,4 @@
+LIBDD_PROFILING {
```

Comment:
This is very nice ! Great idea

Reply:
Yeah, this works pretty well.  Listing only the global exported functions from the .so in CI:
```
$ readelf -Ws --dyn-syms libdd_profiling.so | grep 'GLOBAL DEFAULT' | grep -v UND | grep FUNC
    60: 0000000000003700   736 FUNC    GLOBAL DEFAULT   13 ddprof_start_profiling@@LIBDD_PROFILING
    61: 00000000000039e0   257 FUNC    GLOBAL DEFAULT   13 ddprof_stop_profiling@@LIBDD_PROFILING
   415: 00000000000039e0   257 FUNC    GLOBAL DEFAULT   13 ddprof_stop_profiling
   428: 0000000000003700   736 FUNC    GLOBAL DEFAULT   13 ddprof_start_profiling
```

There are plenty of references to other symbols, but they are marked local.  My understanding is that such symbols are not used at link-time to satisfy unresolved dependencies, which prevents symbol collisions.

Reply:
Yes, this is how I understand it, local symbols won't be used to resolve undefined symbols outside of the library.
I don't think `LIBDD_PROFILING` version is useful, I will remove it.